.github/workflows/sagemaker-integration.yml

```diff
@@ -0,0 +1,148 @@
+        id: create_cpu1
+        run: |
+          cd /home/ubuntu/djl_benchmark_script/scripts
+          token=$( curl -X POST -H "Authorization: token ${{ secrets.ACTION_RUNNER_PERSONAL_TOKEN }}" \
+          https://api.github.com/repos/deepjavalibrary/djl-serving/actions/runners/registration-token \
+          --fail \
+          | jq '.token' | tr -d '"' )
+          ./start_instance.sh action_cpu $token djl-serving
+      - name: Create new CPU instance
+        id: create_cpu2
+        run: |
+          cd /home/ubuntu/djl_benchmark_script/scripts
+          token=$( curl -X POST -H "Authorization: token ${{ secrets.ACTION_RUNNER_PERSONAL_TOKEN }}" \
+          https://api.github.com/repos/deepjavalibrary/djl-serving/actions/runners/registration-token \
+          --fail \
+          | jq '.token' | tr -d '"' )
+          ./start_instance.sh action_cpu $token djl-serving
+    outputs:
+      cpu_instance_id1: ${{ steps.create_cpu1.outputs.action_cpu_instance_id }}
+      cpu_instance_id2: ${{ steps.create_cpu2.outputs.action_cpu_instance_id }}
+
+  # These tests are SLOW, and we only have 2 ml.g5.12xlarge instances available for testing
+  # We parallelize the tests into two separate tracks, each using 1 instance for a few tests.
+  # There's probably a better way to do this, but for now this works.
+  # If you add a test, please try to keep the groups balanced
+  endpoint-tests-group-1:
+    runs-on: [ self-hosted, cpu ]
+    timeout-minutes: 60
+    needs: create-runners
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python3
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10.x'
+      - name: Install pip dependencies
+        run: pip3 install -U boto3 awscli
+      - name: Install SageMaker Python SDK
+        working-directory: tests/integration
+        run: |
+          ./install_sagemaker_pysdk.sh ${{ github.event.inputs.sagemaker-repository }} $ {{ github.event.inputs.repository-branch }}
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: us-east-1
+      - name: Test gpt2xl
+        working-directory: tests/integration
+        run: |
+          python3 llm/sagemaker-endpoint-tests.py gpt2-xl djl
+          echo "sleep 10 seconds to allow endpoint deletion"
+          sleep 10
+      - name: Test stable diffusion
+        if: success() || failure()
+        working-directory: tests/integration
+        run: |
+          python3 llm/sagemaker-endpoint-tests.py stable-diffusion-2-1-base djl
+          echo "sleep 10 seconds to allow endpoint deletion"
+          sleep 10
+      - name: Test opt-1.3b
+        if: success() || failure()
+        working-directory: tests/integration
+        run: |
+          python3 llm/sagemaker-endpoint-tests.py opt-1-3-b djl
+          echo "sleep 10 seconds to allow endpoint deletion"
+          sleep 10
+      - name: Test flan-t5-xxl
```

Comment:
we need a failsafe at the end of all tests. If in any case the endpoint was not shutting down, we need a thanos script to kill everything. Or let's implement an AWS Cops for SageMaker in our lambda function to kill all active port nightly

Reply:
I'll add a script for this, I think we can use a sagemaker.list_endpoints call and delete all the endpoints that are returned.

Reply:
But you don't want to destroy the other endpoint running concurrently right?

Reply:
Right, I think this would be a separate workflow or just a lambda that runs everyday in our account. I don't think having a catch all delete endpoints in this workflow will work for all cases. Sometimes the endpoint stays in the creating stage for more than an hour before it fails. I don't think we can delete endpoints that are in creation phase (even if we know that they will eventually fail).

We need this for multiple accounts/regions, so building something that works for both alpha/gamma in all regions is needed.